lib/foreman_maintain/concerns/base_database.rb

```diff
@@ -22,6 +22,41 @@ def ping(config = configuration)
         psql('SELECT 1 as ping', config)
       end
 
+      def file_path_to_backup_db(config = configuration)
+        dump_file_name = "#{config['database']}_#{Time.now.strftime('%Y-%m-%d_%H-%M-%S')}.dump"
+        "#{db_backup_dir}/#{dump_file_name}.bz2"
+      end
+
+      def backup_db_command(backup_file_path, config = configuration)
+        pg_dump_cmd = "pg_dump -Fc #{config['database']}"
+        "runuser - postgres -c '#{pg_dump_cmd}' | bzip2 -9 > #{backup_file_path}"
```

Comment:
@iNecas,
I would like to know that what difference it makes after extracting out backup file from inner pg_dump command to `runuser`.
I can see that previously, it tried to create dump file with `postgres` user and that's why it was failing with permission error.  After I modification, dump file gets created using `root` user.
Any other differences?

Reply:
The difference is under which user the file gets created: the postgres user doesn't have rights to write to the destination, the root user has.